app/components/specific/elevationProfile/ElevationProfile.js

```diff
@@ -15,22 +16,42 @@ const d3 = {
   d3Array
 };
 
-const ELEVATION_GRADE = {
-  SMALL: 0,
-  EASY: 1,
-  MEDIUM: 2,
-  DIFFICULT: 3,
-  HARD: 4,
-  UNKNOWN: 5
+const getRange = percent => {
+  if (Math.abs(percent) < 5) {
+    return ELEVATION_GRADE.SMALL;
+  } else if (Math.abs(percent) >= 5 && Math.abs(percent) < 7) {
+    return ELEVATION_GRADE.EASY;
+  } else if (Math.abs(percent) >= 7 && Math.abs(percent) < 10) {
+    return ELEVATION_GRADE.MEDIUM;
+  } else if (Math.abs(percent) >= 10 && Math.abs(percent) < 15) {
+    return ELEVATION_GRADE.DIFFICULT;
+  } else if (Math.abs(percent) >= 15) {
+    return ELEVATION_GRADE.HARD;
+  }
+  return ELEVATION_GRADE.UNKNOWN;
+};
+
+const groupEdgesByRange = function(accu, item) {
+  const val = getRange(item.percent);
+  // eslint-disable-next-line no-param-reassign
+  accu[val] = accu[val] || [];
+  accu[val].push(item);
+  return accu;
+};
+
+const groupEdgesBySequence = function(accu, value, index, collection) {
+  if (index > 0 && value.index - collection[index - 1].index === 1) {
+    const group = last(accu);
+    group.push(value);
+  } else {
+    accu.push([value]);
+  }
+  return accu;
 };
 
-const ELEVATION_COLORS = {
-  SMALL: "#f4f6f5",
-  EASY: "#ECBC3E",
-  MEDIUM: "#EA8827",
-  DIFFICULT: "#E1351D",
-  HARD: "#96451F",
-  UNKNOWN: "#00451F"
+// eslint-disable-next-line no-extend-native
+Array.prototype.groupBy = function(fn, accu) {
+  return this.reduce(fn, accu);
```

Comment:
WTF !! that s reduce not groupBy

Reply:
I guess you do not get my point! 
groupEdgesByRange - groupEdgesBySequence functions are doing that job! 
And those functions are groupeBy first parameter.....